api/src/main/resources/workbench.yaml

```diff
@@ -960,6 +960,48 @@ paths:
           description: A list of resources order by last access time desc order
           schema:
             $ref: "#/definitions/RecentResourceResponse"
+
+  /v1/workspaces/{workspaceNamespace}/{workspaceId}/user-recent-resource/updateEntry:
+    parameters:
+      - $ref: '#/parameters/workspaceNamespace'
+      - $ref: '#/parameters/workspaceId'
+    post:
+      tags:
+        - userMetrics
+      description: Add/update Notebook entry to recent resource
+      operationId: updateNotebookEntry
+      parameters:
+        - in: body
+          name: notebook
```

Comment:
I'm not sure what this looks like in JSON. To match other methods I would probably recommend putting a UpdateRecentResourceRequest object here. I recall @UrsaStutsman  hitting some weird swagger codegen bugs when doing something similar to what you have here.